src/execution_impl.rs

```diff
@@ -13,88 +16,119 @@ use wasmer::{
+    let memory = env.wasm_env.memory.get_ref().expect("initialized memory");
+    // TODO: replace all unwrap() by expect()
+    let addr_ptr = StringPtr::new(address as u32);
+    let func_ptr = StringPtr::new(function as u32);
+    let address = &addr_ptr.read(memory).unwrap();
+    let function = &func_ptr.read(memory).unwrap();
+    let value = call_module(env, address, function);
     let ret = StringPtr::alloc(&value.ret, &env.wasm_env).unwrap();
     ret.offset() as i32
 }
 
-/// Create an instance of VM from a module with a
-/// given intefrace, an operation number limit and a webassembly module
+/// `Print` ABI called by the webassembly VM
 ///
-fn create_instance(limit: u64, module: &[u8], interface: &Interface) -> Result<Instance> {
+/// An utility print function to write on stdout directly from AssemblyScript:
+pub fn assembly_script_print(env: &Env, arg: i32) {
+    let str_ptr = StringPtr::new(arg as u32);
+    // TODO: may be in wasmer-as create a pointer that keep memory?
+    println!(
+        "{}",
+        str_ptr
+            .read(env.wasm_env.memory.get_ref().expect("initialized memory"))
+            .unwrap()
+    );
+}
+
+/// Create an instance of VM from a module with a given interface, an operation
+/// number limit and a webassembly module
+pub fn create_instance(limit: u64, module: &[u8], interface: &Interface) -> Result<Instance> {
     let metering = Arc::new(Metering::new(limit, |_: &Operator| -> u64 { 1 }));
     let mut compiler_config = Cranelift::default();
     compiler_config.push_middleware(metering);
     let store = Store::new(&Universal::new(compiler_config).engine());
     let resolver: ImportObject = imports! {
         "env" => {
-            "abort" =>  Function::new_native_with_env(&store, Env::new(interface), abort)
+            "abort" =>  Function::new_native_with_env(&store, Env::new(interface), assembly_script_abort),
         },
-        "index" => {
-            "print" => Function::new_native_with_env(&store, Env::new(interface), print),
-            "call" => Function::new_native_with_env(&store, Env::new(interface), call),
+        "massa" => {
+            "call_module" =>  Function::new_native_with_env(&store, Env::new(interface), assembly_script_call_module),
+            "print" => Function::new_native_with_env(&store, Env::new(interface), assembly_script_print),
         },
     };
     let module = Module::new(&store, &module)?;
     Ok(Instance::new(&module, &resolver)?)
 }
 
-/// fnc: function name
-/// params: function arguments
 pub fn exec(
     limit: u64,
     instance: Option<Instance>,
     module: &[u8],
-    fnc: &str,
-    params: &[i32],
+    function: &str,
+    args: &[i32],
     interface: &Interface,
 ) -> Result<Response> {
     let instance = match instance {
         Some(instance) => instance,
         None => create_instance(limit, module, interface)?,
     };
     let mut p = vec![];
-    for param in params {
+    for param in args {
```

Comment:
param -> arg ?

Reply:
aaaargs :grimacing: 
Do we resolve this? Because here p was kind of useless respectively to the current usage of the PR. It's param again in your PR :beaver: